### DISCUSSION TITLE
FL 21 Shenanigans

### DISCUSSION THREAD

**Discussion by demberto** [demberto/PyFLP#146] [Announcements]
I have come across a few FL 21 projects and it seems Image-Line has garbled up / obfuscated the event values. Most of the values are extremely wrong and out of place. I guess they do not like their beloved format to be parseable.

The basic event structure has not been broken, but the values for **1, 2 and 4 byte events** are entirely unintelligible. Structure-backed event fields are spared from this *yet*.

If anybody is good in figuring out how this can be decoded, you can find my contact details in the README.

**Comment by meint52** [comment]
I want to start getting into learning about the .flp file structure. I have little background knowledge. What would be the best place to start?

**Reply by demberto** [reply]
I believe I have documented the FLP format adequately in PyFLP's docs. You can begin here: https://pyflp.readthedocs.io/en/latest/architecture/flp-format.html

**Reply by meint52** [reply]
Thanks a lot! I really like this project and i will definitely dive into the details. Although it will be quite a challange for me as i know nothing about binary... I come from bioinformatics. I made a script that finds all FLP's stored on your computer (besides autosaves, backups and project demo's) and turns them into ziploops stored into a specified folder. The idea is that people can easily backup all their projects with the click of a button. (Usefull for chaotic people like me) For most projects it works but for some it runs into the same issue as #140. Hopefully this motivates me enough to figure out more about the flp format.

**Reply by c1bles** [reply]
Hi! Love your project and all but have you found a fix? I **can't** buy fl studio rn and i made some fire songs but closed the project by accident... I don't really have friends too so i can't do the "send to a friend who owns the program" method. Thanks in advance!
